lesson11/js/main.js

```diff
@@ -0,0 +1,70 @@
+"use strict";
+const menuContainer = document.getElementById("js-menu_container");
+const menuLists = document.createElement("ul");
+const loading = document.getElementById("js-loading");
+
+function showLoadingImg() {
+  const loadingImg = document.createElement("img");
+  loadingImg.src = "images/loading-circle.gif";
+  loadingImg.classList.add("el_loadingImg");
+
+  loading.appendChild(loadingImg);
+}
+
+function createMenuList(values) {
+  const fragment = document.createDocumentFragment();
+
+  Object.keys(values).forEach((key) => {
+    values[key].forEach(value => {
+      const listItem = document.createElement("li");
+  
+      const listImg = document.createElement("img");
+      listImg.src = value.img;
+      listImg.alt = value.alt;
+      listImg.width = 40;
+      listImg.height = 40;
+  
+      const listAnchor = document.createElement("a");
+      listAnchor.href = value.a;
+      listAnchor.textContent = value.text;
+  
+      fragment.appendChild(listItem).appendChild(listAnchor).appendChild(listImg);
+
+    })
+  });
+
+  menuContainer.appendChild(menuLists).appendChild(fragment);
+}
+
+async function fetchMenuContents(){
+  showLoadingImg();
+  await new Promise((resolve) => setTimeout(resolve, 3000));
```

Comment:
Please check this implementation.
I added a setTimeout of 3 seconds before the fetch process, but I'm worried if this code is okay.

Reply:
I reprint the comments I received from the doctor when he reviewed my work.

> setTimeoutはこの課題では不要です。setTimeoutは以前の課題までfetchしている間待つために擬似でやっていました。今はjsonデータを実際にリクエストして取っているのでここは不要です

So, there is no problem deleting this function.

Reply:
Thank you for sharing.

I removed setTimeout with this commit.

[6786034](https://github.com/nyattsu72/hanson-frontend-cource/pull/14/commits/67860346f836519c6ec6ed9a3360b63cb2436d95)